

<!DOCTYPE html>


<html lang="en" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

    <title>Analyser la qualité avec fastQC &#8212; Linux en Bioinformatique</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "light";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=365ca57ee442770a23c6" rel="stylesheet" />
<link href="_static/styles/bootstrap.css?digest=365ca57ee442770a23c6" rel="stylesheet" />
<link href="_static/styles/pydata-sphinx-theme.css?digest=365ca57ee442770a23c6" rel="stylesheet" />

  
  <link href="_static/vendor/fontawesome/6.1.2/css/all.min.css?digest=365ca57ee442770a23c6" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" href="_static/styles/sphinx-book-theme.css?digest=14f4ca6b54d191a8c7657f6c759bf11a5fb86285" type="text/css" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="_static/design-style.4045f2051d55cab465a707391d5b2007.min.css" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/bootstrap.js?digest=365ca57ee442770a23c6" />
<link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=365ca57ee442770a23c6" />
  <script src="_static/vendor/fontawesome/6.1.2/js/all.min.js?digest=365ca57ee442770a23c6"></script>

    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script src="_static/scripts/sphinx-book-theme.js?digest=5a5c038af52cf7bc1a1ec88eea08e6366ee68824"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="_static/sphinx-thebe.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = '13_fastqc';</script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Rogner les reads" href="14_rogner.html" />
    <link rel="prev" title="Jeu de données" href="12_data.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <a class="skip-link" href="#main-content">Skip to main content</a>
  
  <div id="pst-scroll-pixel-helper"></div>

  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>
    Back to top
  </button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__primary"
          id="__primary"/>
  <label class="overlay overlay-primary" for="__primary"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__secondary"
          id="__secondary"/>
  <label class="overlay overlay-secondary" for="__secondary"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>
  
    <nav class="bd-header navbar navbar-expand-lg bd-navbar">
    </nav>
  
  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">
  

<a class="navbar-brand logo" href="00_home.html">
  
  
  
  
  
    
    
      
    
    
    <img src="_static/logo.png" class="logo__image only-light" alt="Linux en Bioinformatique - Home"/>
    <script>document.write(`<img src="_static/logo.png" class="logo__image only-dark" alt="Linux en Bioinformatique - Home"/>`);</script>
  
  
</a></div>
        <div class="sidebar-primary-item"><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="00_home.html">
                    A la découverte de Linux
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Introduction</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="01_team.html">L’équipe pédagogique</a></li>
<li class="toctree-l1"><a class="reference internal" href="02_introduction.html">Le cluster de l’IFB</a></li>
<li class="toctree-l1"><a class="reference internal" href="03_jupyter.html">C’est quoi JupyterHub</a></li>
<li class="toctree-l1"><a class="reference internal" href="04_bash.html">Comment converser avec le terminal ?</a></li>
<li class="toctree-l1"><a class="reference internal" href="05_module.html">Charger l’environnement logiciel</a></li>
<li class="toctree-l1"><a class="reference internal" href="06_protoCommande.html">Prototype(s) d’une commande</a></li>
<li class="toctree-l1"><a class="reference internal" href="quizz_01.html">Quizz 1</a></li>
<li class="toctree-l1"><a class="reference internal" href="cards_01.html">Fiches de révision</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Arborescence</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="07_arborescence.html">Arborescence de fichiers</a></li>
<li class="toctree-l1"><a class="reference internal" href="08_storage.html">Stockage fichiers</a></li>
<li class="toctree-l1"><a class="reference internal" href="09_manipulate.html">Manipuler des fichiers</a></li>
<li class="toctree-l1"><a class="reference internal" href="10_redirection.html">Redirections</a></li>
<li class="toctree-l1"><a class="reference internal" href="11_bonnesPratiques.html">Quelques bonnes pratiques…</a></li>
<li class="toctree-l1"><a class="reference internal" href="quizz_02.html">Quizz 2</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Qualité des lectures</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="12_data.html">Jeu de données</a></li>
<li class="toctree-l1 current active"><a class="current reference internal" href="#">Analyser la qualité avec fastQC</a></li>
<li class="toctree-l1"><a class="reference internal" href="14_rogner.html">Rogner les reads</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Read mapping</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="15_mapping.html">Mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="16_stopJupyter.html">Arrêter le serveur Jupyterhub</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Ressources</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="17_raccourcis.html">Les raccourcis clavier du terminal</a></li>
<li class="toctree-l1"><a class="reference internal" href="formats.html">Formats</a></li>
<li class="toctree-l1"><a class="reference internal" href="markdown.html">Markdown</a></li>

<li class="toctree-l1"><a class="reference internal" href="pourquoiCaPlante.html">Pourquoi ça plante ?</a></li>
<li class="toctree-l1"><a class="reference internal" href="glossaire.html">Glossaire</a></li>
<li class="toctree-l1"><a class="reference internal" href="genindex.html">Index</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><label class="sidebar-toggle primary-toggle btn btn-sm" for="__primary" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</label></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/IFB-ElixirFr/LinuxEBAII" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/IFB-ElixirFr/LinuxEBAII/issues/new?title=Issue%20on%20page%20%2F13_fastqc.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="_sources/13_fastqc.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>


<script>
document.write(`
  <button class="btn btn-sm navbar-btn theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="theme-switch nav-link" data-mode="light"><i class="fa-solid fa-sun fa-lg"></i></span>
    <span class="theme-switch nav-link" data-mode="dark"><i class="fa-solid fa-moon fa-lg"></i></span>
    <span class="theme-switch nav-link" data-mode="auto"><i class="fa-solid fa-circle-half-stroke fa-lg"></i></span>
  </button>
`);
</script>

<script>
document.write(`
  <button class="btn btn-sm navbar-btn search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<label class="sidebar-toggle secondary-toggle btn btn-sm" for="__secondary"title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</label>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Analyser la qualité avec fastQC</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#jupyter-lab-acces-au-fichier-html">Jupyter Lab : accès au fichier html</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#telecharger-les-resultats-avec-cyberduck-osx">Télécharger les résultats avec Cyberduck (OSX)</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#resultats">Resultats</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#interpretation">Interprétation</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#basic-statistics">Basic Statistics</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#per-base-sequence-quality">Per Base Sequence Quality</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#per-tile-sequence-quality">Per Tile Sequence Quality</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#per-sequence-quality-scores">Per Sequence Quality Scores</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#per-base-sequence-content">Per Base Sequence Content</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#per-sequence-gc-content">Per Sequence GC Content</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#per-base-n-content">Per Base N Content</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#sequence-length-distribution">Sequence Length Distribution</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#duplicate-sequences">Duplicate Sequences</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#overrepresented-sequences">Overrepresented Sequences</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#adapter-content">Adapter Content</a></li>
</ul>
</li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article" role="main">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="analyser-la-qualite-avec-fastqc">
<h1>Analyser la qualité avec fastQC<a class="headerlink" href="#analyser-la-qualite-avec-fastqc" title="Permalink to this heading">#</a></h1>
<p>Fast Quality Control (FastQC)</p>
<p>Propose un certain nombre de diagrammes pour évaluer la qualité du séquençage.</p>
<p><code class="docutils literal notranslate"><span class="pre">fastqc</span> <span class="pre">[-o</span> <span class="pre">output</span> <span class="pre">dir]</span> <span class="pre">[--(no)extract]</span> <span class="pre">[-f</span> <span class="pre">fastq|bam|sam]</span> <span class="pre">fq1</span> <span class="pre">fq2</span> <span class="pre">...</span></code></p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># On remonte d’un niveau dans l’arborescence</span>
<span class="nb">cd</span><span class="w"> </span>/shared/projects/2325_ebaii/coursLinux/demo/chip-seq/<span class="w">                                   </span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>?2004h
</pre></div>
</div>
<div class="output traceback highlight-ipythontb notranslate"><div class="highlight"><pre><span></span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># On créé un répertoire</span>
mkdir<span class="w"> </span>-p<span class="w"> </span>qc
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># 2 instructions sur la même ligne séparées par ‘;’</span>
ls<span class="w"> </span>-l<span class="w"> </span><span class="p">;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>qc
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>total 16
drwxrwx---+ 2 pfrancois pfrancois 4096 Nov  4 13:05 <span class=" -Color -Color-Bold -Color-Bold-Blue">fastq</span>
drwxrwx---+ 3 pfrancois pfrancois 4096 Nov  4 13:03 <span class=" -Color -Color-Bold -Color-Bold-Blue">output</span>
drwxrwx---+ 2 pfrancois pfrancois 4096 Nov  4 13:05 <span class=" -Color -Color-Bold -Color-Bold-Blue">qc</span>
drwxrwx---+ 2 pfrancois pfrancois 4096 Nov  4 13:03 <span class=" -Color -Color-Bold -Color-Bold-Blue">ref</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># Charge l&#39;outil fastqc dans l’environnement  </span>
module<span class="w"> </span>load<span class="w"> </span>fastqc/0.11.8<span class="w"> </span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># Obtenir de l’aide </span>
fastqc<span class="w"> </span>-h<span class="w"> </span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>            FastQC - A high throughput sequence QC analysis tool

SYNOPSIS

	fastqc seqfile1 seqfile2 .. seqfileN

    fastqc [-o output dir] [--(no)extract] [-f fastq|bam|sam] 
           [-c contaminant file] seqfile1 .. seqfileN

DESCRIPTION

    FastQC reads a set of sequence files and produces from each one a quality
    control report consisting of a number of different modules, each one of 
    which will help to identify a different potential type of problem in your
    data.
    
    If no files to process are specified on the command line then the program
    will start as an interactive graphical application.  If files are provided
    on the command line then the program will run with no user interaction
    required.  In this mode it is suitable for inclusion into a standardised
    analysis pipeline.
    
    The options for the program as as follows:
    
    -h --help       Print this help file and exit
    
    -v --version    Print the version of the program and exit
    
    -o --outdir     Create all output files in the specified output directory.
                    Please note that this directory must exist as the program
                    will not create it.  If this option is not set then the 
                    output file for each sequence file is created in the same
                    directory as the sequence file which was processed.
                    
    --casava        Files come from raw casava output. Files in the same sample
                    group (differing only by the group number) will be analysed
                    as a set rather than individually. Sequences with the filter
                    flag set in the header will be excluded from the analysis.
                    Files must have the same names given to them by casava
                    (including being gzipped and ending with .gz) otherwise they
                    won&#39;t be grouped together correctly.
                    
    --nano          Files come from nanopore sequences and are in fast5 format. In
                    this mode you can pass in directories to process and the program
                    will take in all fast5 files within those directories and produce
                    a single output file from the sequences found in all files.                    
                    
    --nofilter      If running with --casava then don&#39;t remove read flagged by
                    casava as poor quality when performing the QC analysis.
                   
    --extract       If set then the zipped output file will be uncompressed in
                    the same directory after it has been created.  By default
                    this option will be set if fastqc is run in non-interactive
                    mode.
                    
    -j --java       Provides the full path to the java binary you want to use to
                    launch fastqc. If not supplied then java is assumed to be in
                    your path.
                   
    --noextract     Do not uncompress the output file after creating it.  You
                    should set this option if you do not wish to uncompress
                    the output when running in non-interactive mode.
                    
    --nogroup       Disable grouping of bases for reads &gt;50bp. All reports will
                    show data for every base in the read.  WARNING: Using this
                    option will cause fastqc to crash and burn if you use it on
                    really long reads, and your plots may end up a ridiculous size.
                    You have been warned!
                    
    --min_length    Sets an artificial lower limit on the length of the sequence
                    to be shown in the report.  As long as you set this to a value
                    greater or equal to your longest read length then this will be
                    the sequence length used to create your read groups.  This can
                    be useful for making directly comaparable statistics from 
                    datasets with somewhat variable read lengths.
                    
    -f --format     Bypasses the normal sequence file format detection and
                    forces the program to use the specified format.  Valid
                    formats are bam,sam,bam_mapped,sam_mapped and fastq
                    
    -t --threads    Specifies the number of files which can be processed
                    simultaneously.  Each thread will be allocated 250MB of
                    memory so you shouldn&#39;t run more threads than your
                    available memory will cope with, and not more than
                    6 threads on a 32 bit machine
                  
    -c              Specifies a non-default file which contains the list of
    --contaminants  contaminants to screen overrepresented sequences against.
                    The file must contain sets of named contaminants in the
                    form name[tab]sequence.  Lines prefixed with a hash will
                    be ignored.

    -a              Specifies a non-default file which contains the list of
    --adapters      adapter sequences which will be explicity searched against
                    the library. The file must contain sets of named adapters
                    in the form name[tab]sequence.  Lines prefixed with a hash
                    will be ignored.
                    
    -l              Specifies a non-default file which contains a set of criteria
    --limits        which will be used to determine the warn/error limits for the
                    various modules.  This file can also be used to selectively 
                    remove some modules from the output all together.  The format
                    needs to mirror the default limits.txt file found in the
                    Configuration folder.
                    
   -k --kmers       Specifies the length of Kmer to look for in the Kmer content
                    module. Specified Kmer length must be between 2 and 10. Default
                    length is 7 if not specified.
                    
   -q --quiet       Supress all progress messages on stdout and only report errors.
   
   -d --dir         Selects a directory to be used for temporary files written when
                    generating report images. Defaults to system temp directory if
                    not specified.
                    
BUGS

    Any bugs in fastqc should be reported either to simon.andrews@babraham.ac.uk
    or in www.bioinformatics.babraham.ac.uk/bugzilla/
                   
    
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># Lancer fastqc </span>
fastqc<span class="w"> </span>-f<span class="w"> </span>fastq<span class="w"> </span>-o<span class="w"> </span>./<span class="w"> </span>../fastq/siNT_ER_E2_r3_chr21.fastq<span class="w"> </span><span class="m">2</span>&gt;<span class="w"> </span>siNT_ER_E2_r3_chr21_fastqc.log
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Analysis complete for siNT_ER_E2_r3_chr21.fastq
</pre></div>
</div>
</div>
</div>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>less<span class="w"> </span>siNT_ER_E2_r3_chr21_fastqc.log<span class="w">  </span>
</pre></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># Que voyez vous ?</span>
ls<span class="w"> </span>
</pre></div>
</div>
</div>
</div>
<section id="jupyter-lab-acces-au-fichier-html">
<h2>Jupyter Lab : accès au fichier html<a class="headerlink" href="#jupyter-lab-acces-au-fichier-html" title="Permalink to this heading">#</a></h2>
<p>Côté gauche, avec l’onglet “répertoire” se placer à la racine du cluster</p>
<p>Sélectionner les répertoires jusqu’au répertoire de travail <code class="docutils literal notranslate"><span class="pre">/shared/projects/&lt;project&gt;/chip-seq/qc</span></code> (adapter <code class="docutils literal notranslate"><span class="pre">&lt;projet&gt;</span></code>)</p>
<p>Cliquer sur le fichier html pour l’ouvrir dans un nouvel onglet</p>
<p><img alt="" src="_images/ebaii_acces_html.png" /></p>
</section>
<section id="telecharger-les-resultats-avec-cyberduck-osx">
<h2>Télécharger les résultats avec Cyberduck (OSX)<a class="headerlink" href="#telecharger-les-resultats-avec-cyberduck-osx" title="Permalink to this heading">#</a></h2>
<p><img alt="" src="_images/ebaii_cyberduck1.png" /></p>
<p><img alt="" src="_images/ebaii_cyberduck2.png" /></p>
<p><img alt="" src="_images/ebaii_cyberduck3.png" /></p>
<p><img alt="" src="_images/ebaii_cyberduck4.png" /></p>
<p><img alt="" src="_images/ebaii_cyberduck5.png" /></p>
</section>
<section id="resultats">
<h2>Resultats<a class="headerlink" href="#resultats" title="Permalink to this heading">#</a></h2>
<p>Les résultats sont disponibles <a href='data/siNT_ER_E2_r3_chr21_fastqc.html' target='_blank'> ici </a>.</p>
<iframe src="data/siNT_ER_E2_r3_chr21_fastqc.html" name="resultsIframe"
        height="500" width="100%" marginwidth="50" scrolling="yes"> ></iframe>
</section>
<section id="interpretation">
<h2>Interprétation<a class="headerlink" href="#interpretation" title="Permalink to this heading">#</a></h2>
<p>Exploration des résultats de fastqc en interactif.</p>
<ul class="simple">
<li><p>A quoi correspond le diagramme “Per base sequence quality”.</p></li>
<li><p>A quoi correspond le diagramme “Per sequence quality score” ?</p></li>
<li><p>A quoi correspond le diagramme “Per base sequence content” ?</p></li>
<li><p>A quoi correspond le diagramme “Per sequence GC content” ?</p></li>
<li><p>A quoi correspond le diagramme “Per sequence N content” ?</p></li>
<li><p>A quoi correspond le diagramme “Sequence length distribution” ?</p></li>
<li><p>A quoi correspond le diagramme “Sequence duplication level” ?</p></li>
<li><p>A quoi correspond le diagramme “Kmer content” ?</p></li>
</ul>
<p><img alt="" src="_images/ebaii_fastqc_tricolore.png" /></p>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<p>Les explications ci-dessous sont la compilation des 3 ressources suivantes :</p>
<ul class="simple">
<li><p><a class="reference external" href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/">https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/</a></p></li>
<li><p><a class="reference external" href="https://dnacore.missouri.edu/PDF/FastQC_Manual.pdf">https://dnacore.missouri.edu/PDF/FastQC_Manual.pdf</a></p></li>
<li><p><a class="reference external" href="https://rtsf.natsci.msu.edu/genomics/tech-notes/fastqc-tutorial-and-faq/">https://rtsf.natsci.msu.edu/genomics/tech-notes/fastqc-tutorial-and-faq/</a></p></li>
</ul>
</div>
<section id="basic-statistics">
<h3>Basic Statistics<a class="headerlink" href="#basic-statistics" title="Permalink to this heading">#</a></h3>
<p><img alt="" src="_images/ebaii_fastqc_stat.png" /></p>
<p>The Basic Statistics module generates some simple composition statistics for the file analysed.</p>
<ul class="simple">
<li><p>Filename</p></li>
<li><p>File type</p></li>
<li><p>Encoding: Says which ASCII encoding of quality values was found in this file.</p></li>
<li><p>Total Sequences: A count of the total number of sequences processed.</p></li>
<li><p>Filtered Sequences:</p></li>
<li><p>Sequence Length</p></li>
<li></li>
</ul>
<div class="attention admonition">
<p class="admonition-title">Warning</p>
<p>Basic Statistics never raises a warning.</p>
</div>
<div class="error admonition">
<p class="admonition-title">Error</p>
<p>Basic Statistics never raises an error.</p>
</div>
</section>
<section id="per-base-sequence-quality">
<h3>Per Base Sequence Quality<a class="headerlink" href="#per-base-sequence-quality" title="Permalink to this heading">#</a></h3>
<div class="sd-container-fluid sd-sphinx-override sd-mb-4 docutils">
<div class="sd-row sd-g-2 sd-g-xs-2 sd-g-sm-2 sd-g-md-2 sd-g-lg-2 docutils">
<div class="sd-col sd-d-flex-column docutils">
<p><img alt="" src="_images/ebaii_fastqc_pbsq_ok.png" /></p>
</div>
<div class="sd-col sd-d-flex-column docutils">
<p><img alt="" src="_images/ebaii_fastqc_pbsq_no.png" /></p>
</div>
</div>
</div>
<p>This view shows an overview of the range of quality values across all bases at each position in the FastQ file.
For each position a BoxWhisker type plot is drawn. The elements of the plot are as follows:  The central red line is the median value  The yellow box represents the inter-quartile range (25-75%)  The upper and lower whiskers represent the 10% and 90% points  The blue line represents the mean quality The y-axis on the graph shows the quality scores. The higher the score the better the base call. The background of the graph divides the y axis into very good quality calls (green), calls of reasonable quality (orange), and calls of poor quality (red). The quality of calls on most platforms will degrade as the run progresses, so it is common to see base calls falling into the orange area towards the end of a read.</p>
<div class="attention admonition">
<p class="admonition-title">Warning</p>
<p>A warning will be issued if the lower quartile for any base is less than 10, or if the median for any base is less than 25.</p>
</div>
<div class="error admonition">
<p class="admonition-title">Error</p>
<p>This module will raise a failure if the lower quartile for any base is less than 5 or if the median for any base is less than 20.</p>
</div>
</section>
<section id="per-tile-sequence-quality">
<h3>Per Tile Sequence Quality<a class="headerlink" href="#per-tile-sequence-quality" title="Permalink to this heading">#</a></h3>
<div class="sd-container-fluid sd-sphinx-override sd-mb-4 docutils">
<div class="sd-row sd-g-2 sd-g-xs-2 sd-g-sm-2 sd-g-md-2 sd-g-lg-2 docutils">
<div class="sd-col sd-d-flex-column docutils">
<p><img alt="" src="_images/ebaii_fastqc_ptsq_ok.png" /></p>
</div>
<div class="sd-col sd-d-flex-column docutils">
<p><img alt="" src="_images/ebaii_fastqc_ptsq_no.png" /></p>
</div>
</div>
</div>
<p>The plot shows the deviation from the average quality for each tile. The colours are on a cold to hot scale, with cold colours being positions where the quality was at or above the average for that base in the run, and hotter colours indicate that a tile had worse qualities than other tiles for that base. In the example below you can see that certain tiles show consistently poor quality. A good plot should be blue all over.</p>
<div class="attention admonition">
<p class="admonition-title">Warning</p>
<p>This module will issue a warning if any tile shows a mean Phred score more than 2 less than the mean for that base across all tiles.</p>
</div>
<div class="error admonition">
<p class="admonition-title">Error</p>
<p>This module will issue a warning if any tile shows a mean Phred score more than 5 less than the mean for that base across all tiles.</p>
</div>
</section>
<section id="per-sequence-quality-scores">
<h3>Per Sequence Quality Scores<a class="headerlink" href="#per-sequence-quality-scores" title="Permalink to this heading">#</a></h3>
<div class="sd-container-fluid sd-sphinx-override sd-mb-4 docutils">
<div class="sd-row sd-g-2 sd-g-xs-2 sd-g-sm-2 sd-g-md-2 sd-g-lg-2 docutils">
<div class="sd-col sd-d-flex-column docutils">
<p><img alt="" src="_images/ebaii_fastqc_psqs_ok.png" /></p>
</div>
<div class="sd-col sd-d-flex-column docutils">
<p><img alt="" src="_images/ebaii_fastqc_psqs_no.png" /></p>
</div>
</div>
</div>
<p>The per sequence quality score report allows you to see if a subset of your sequences have universally low quality values. It is often the case that a subset of sequences will have universally poor quality, often because they are poorly imaged (on the edge of the field of view etc), however these should represent only a small percentage of the total sequences.</p>
<p>If a significant proportion of the sequences in a run have overall low quality then this could indicate some kind of systematic problem - possibly with just part of the run (for example one end of a flowcell).</p>
<div class="attention admonition">
<p class="admonition-title">Warning</p>
<p>A warning is raised if the most frequently observed mean quality is below 27 - this equates to a 0.2% error rate.</p>
</div>
<div class="error admonition">
<p class="admonition-title">Error</p>
<p>An error is raised if the most frequently observed mean quality is below 20 - this equates to a 1% error rate.</p>
</div>
</section>
<section id="per-base-sequence-content">
<h3>Per Base Sequence Content<a class="headerlink" href="#per-base-sequence-content" title="Permalink to this heading">#</a></h3>
<div class="sd-container-fluid sd-sphinx-override sd-mb-4 docutils">
<div class="sd-row sd-g-2 sd-g-xs-2 sd-g-sm-2 sd-g-md-2 sd-g-lg-2 docutils">
<div class="sd-col sd-d-flex-column docutils">
<p><img alt="" src="_images/ebaii_fastqc_pbsc_ok.png" /></p>
</div>
<div class="sd-col sd-d-flex-column docutils">
<p><img alt="" src="_images/ebaii_fastqc_pbsc_no.png" /></p>
</div>
</div>
</div>
<p>Per Base Sequence Content plots out the proportion of each base position in a file for which each of the four normal DNA bases has been called.</p>
<p>In a random library you would expect that there would be little to no difference between the different bases of a sequence run, so the lines in this plot should run parallel with each other.</p>
<p>If you see strong biases which change in different bases then this usually indicates an overrepresented sequence which is contaminating your library. A bias which is consistent across all bases either indicates that the original library was sequence biased, or that there was a systematic problem during the sequencing of the library.</p>
<div class="attention admonition">
<p class="admonition-title">Warning</p>
<p>This module issues a warning if the difference between A and T, or G and C is greater than 10% in any position.</p>
</div>
<div class="error admonition">
<p class="admonition-title">Error</p>
<p>This module will fail if the difference between A and T, or G and C is greater than 20% in any position.</p>
</div>
</section>
<section id="per-sequence-gc-content">
<h3>Per Sequence GC Content<a class="headerlink" href="#per-sequence-gc-content" title="Permalink to this heading">#</a></h3>
<div class="sd-container-fluid sd-sphinx-override sd-mb-4 docutils">
<div class="sd-row sd-g-2 sd-g-xs-2 sd-g-sm-2 sd-g-md-2 sd-g-lg-2 docutils">
<div class="sd-col sd-d-flex-column docutils">
<p><img alt="" src="_images/ebaii_fastqc_psGCc_ok.png" /></p>
</div>
<div class="sd-col sd-d-flex-column docutils">
<p><img alt="" src="_images/ebaii_fastqc_psGCc_no.png" /></p>
</div>
</div>
</div>
<p>This module measures the GC content across the whole length of each sequence in a file and compares it to a modelled normal distribution of GC content.
In a normal random library you would expect to see a roughly normal distribution of GC content where the central peak corresponds to the overall GC content of the underlying genome. Since we don’t know the the GC content of the genome the modal GC content is calculated from the observed data and used to build a reference distribution. An unusually shaped distribution could indicate a contaminated library or some other kinds of biased subset. A normal distribution which is shifted indicates some systematic bias which is independent of base position. If there is a systematic bias which creates a shifted normal distribution then this won’t be flagged as an error by the module since it doesn’t know what your genome’s GC content should be.</p>
<div class="attention admonition">
<p class="admonition-title">Warning</p>
<p>A warning is raised if the sum of the deviations from the normal distribution represents more than 15% of the reads.</p>
</div>
<div class="error admonition">
<p class="admonition-title">Error</p>
<p>This module will indicate a failure if the sum of the deviations from the normal distribution represents more than 30% of the reads.</p>
</div>
</section>
<section id="per-base-n-content">
<h3>Per Base N Content<a class="headerlink" href="#per-base-n-content" title="Permalink to this heading">#</a></h3>
<div class="sd-container-fluid sd-sphinx-override sd-mb-4 docutils">
<div class="sd-row sd-g-2 sd-g-xs-2 sd-g-sm-2 sd-g-md-2 sd-g-lg-2 docutils">
<div class="sd-col sd-d-flex-column docutils">
<p><img alt="" src="_images/ebaii_fastqc_pbNc_ok.png" /></p>
</div>
<div class="sd-col sd-d-flex-column docutils">
<p><img alt="" src="_images/ebaii_fastqc_pbNc_no.png" /></p>
</div>
</div>
</div>
<p>If a sequencer is unable to make a base call with sufficient confidence then it will normally substitute an N rather than a conventional base call This module plots out the percentage of base calls at each position for which an N was called.</p>
<p>It’s not unusual to see a very low proportion of Ns appearing in a sequence, especially nearer the end of a sequence. However, if this proportion rises above a few percent it suggests that the analysis pipeline was unable to interpret the data well enough to make valid base calls.</p>
<div class="attention admonition">
<p class="admonition-title">Warning</p>
<p>This module raises a warning if any position shows an N content of &gt;5%.</p>
</div>
<div class="error admonition">
<p class="admonition-title">Error</p>
<p>This module will raise an error if any position shows an N content of &gt;20%.</p>
</div>
</section>
<section id="sequence-length-distribution">
<h3>Sequence Length Distribution<a class="headerlink" href="#sequence-length-distribution" title="Permalink to this heading">#</a></h3>
<div class="sd-container-fluid sd-sphinx-override sd-mb-4 docutils">
<div class="sd-row sd-g-2 sd-g-xs-2 sd-g-sm-2 sd-g-md-2 sd-g-lg-2 docutils">
<div class="sd-col sd-d-flex-column docutils">
<p><img alt="" src="_images/fastqc_sld_ok.png" /></p>
</div>
<div class="sd-col sd-d-flex-column docutils">
<p><img alt="" src="_images/fastqc_sld_bad.png" /></p>
</div>
</div>
</div>
<p>Some high throughput sequencers generate sequence fragments of uniform length, but others can contain reads of wildly varying lengths. Even within uniform length libraries some pipelines will trim sequences to remove poor quality base calls from the end. This module generates a graph showing the distribution of fragment sizes in the file which was analysed.
In many cases this will produce a simple graph showing a peak only at one size, but for variable length FastQ files this will show the relative amounts of each different size of sequence fragment.</p>
<div class="attention admonition">
<p class="admonition-title">Warning</p>
<p>This module will raise a warning if all sequences are not the same length.</p>
</div>
<div class="error admonition">
<p class="admonition-title">Error</p>
<p>This module will raise an error if any of the sequences have zero length.</p>
</div>
</section>
<section id="duplicate-sequences">
<h3>Duplicate Sequences<a class="headerlink" href="#duplicate-sequences" title="Permalink to this heading">#</a></h3>
<div class="sd-container-fluid sd-sphinx-override sd-mb-4 docutils">
<div class="sd-row sd-g-2 sd-g-xs-2 sd-g-sm-2 sd-g-md-2 sd-g-lg-2 docutils">
<div class="sd-col sd-d-flex-column docutils">
<p><img alt="" src="_images/ebaii_fastqc_duplicSeq_ok.png" /></p>
</div>
<div class="sd-col sd-d-flex-column docutils">
<p><img alt="" src="_images/ebaii_fastqc_duplicSeq_RNA.png" /></p>
</div>
</div>
</div>
<p>In a diverse library most sequences will occur only once in the final set. A low level of duplication may indicate a very high level of coverage of the target sequence, but a high level of duplication is more likely to indicate some kind of enrichment bias (eg PCR over amplification).</p>
<p>This module counts the degree of duplication for every sequence in the set and creates a plot showing the relative number of sequences with different degrees of duplication.</p>
<div class="attention admonition">
<p class="admonition-title">Warning</p>
<p>This module will issue a warning if non-unique sequences make up more than 20% of the total.</p>
</div>
<div class="error admonition">
<p class="admonition-title">Error</p>
<p>This module will issue a error if non-unique sequences make up more than 50% of the total.</p>
</div>
</section>
<section id="overrepresented-sequences">
<h3>Overrepresented Sequences<a class="headerlink" href="#overrepresented-sequences" title="Permalink to this heading">#</a></h3>
<p><img alt="" src="_images/ebaii_fastqc_overSeq_polyA.png" /></p>
<p>A normal high-throughput library will contain a diverse set of sequences, with no individual sequence making up a tiny fraction of the whole. Finding that a single sequence is very overrepresented in the set either means that it is highly biologically significant, or indicates that the library is contaminated, or not as diverse as you expected. This module lists all of the sequence which make up more than 0.1% of the total. To conserve memory only sequences which appear in the first 200,000 sequences are tracked to the end of the file. It is therefore possible that a sequence which is overrepresented but doesn’t appear at the start of the file for some reason could be missed by this module.</p>
<div class="attention admonition">
<p class="admonition-title">Warning</p>
<p>This module will issue a warning if any sequence is found to represent more than 0.1% of the total.</p>
</div>
<div class="error admonition">
<p class="admonition-title">Error</p>
<p>This module will issue an error if any sequence is found to represent more than 1% of the total.</p>
</div>
</section>
<section id="adapter-content">
<h3>Adapter Content<a class="headerlink" href="#adapter-content" title="Permalink to this heading">#</a></h3>
<div class="sd-container-fluid sd-sphinx-override sd-mb-4 docutils">
<div class="sd-row sd-g-2 sd-g-xs-2 sd-g-sm-2 sd-g-md-2 sd-g-lg-2 docutils">
<div class="sd-col sd-d-flex-column docutils">
<p><img alt="" src="_images/ebaii_fastqc_adaptor_ok.png" /></p>
</div>
<div class="sd-col sd-d-flex-column docutils">
<p><img alt="" src="_images/ebaii_fastqc_adaptor_litle.png" /></p>
</div>
</div>
</div>
<p>The Kmer Content module will do a generic analysis of all of the Kmers in your library to find those which do not have even coverage through the length of your reads. This can find a number of different sources of bias in the library which can include the presence of read-through adapter sequences building up on the end of your sequences.
You can however find that the presence of any overrepresented sequences in your library (such as adapter dimers) will cause the Kmer plot to be dominated by the Kmers these sequences contain, and that it’s not always easy to see if there are other biases present in which you might be interested.</p>
<div class="attention admonition">
<p class="admonition-title">Warning</p>
<p>This module will issue a warning if any sequence is present in more than 5% of all reads.</p>
</div>
<div class="error admonition">
<p class="admonition-title">Error</p>
<p>This module will issue a warning if any sequence is present in more than 10% of all reads.</p>
</div>
</section>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "bash"
        },
        kernelOptions: {
            name: "bash",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'bash'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer">
                  <!-- Previous / next buttons -->
<div class="prev-next-area">
    <a class="left-prev"
       href="12_data.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Jeu de données</p>
      </div>
    </a>
    <a class="right-next"
       href="14_rogner.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Rogner les reads</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">

  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#jupyter-lab-acces-au-fichier-html">Jupyter Lab : accès au fichier html</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#telecharger-les-resultats-avec-cyberduck-osx">Télécharger les résultats avec Cyberduck (OSX)</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#resultats">Resultats</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#interpretation">Interprétation</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#basic-statistics">Basic Statistics</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#per-base-sequence-quality">Per Base Sequence Quality</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#per-tile-sequence-quality">Per Tile Sequence Quality</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#per-sequence-quality-scores">Per Sequence Quality Scores</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#per-base-sequence-content">Per Base Sequence Content</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#per-sequence-gc-content">Per Sequence GC Content</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#per-base-n-content">Per Base N Content</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#sequence-length-distribution">Sequence Length Distribution</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#duplicate-sequences">Duplicate Sequences</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#overrepresented-sequences">Overrepresented Sequences</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#adapter-content">Adapter Content</a></li>
</ul>
</li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Thomas Denecker & Claire Toffano-Nioche
</p>

  </div>
  
  <div class="footer-item">
    
  <p class="copyright">
    
      © Copyright 2022.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="_static/scripts/bootstrap.js?digest=365ca57ee442770a23c6"></script>
<script src="_static/scripts/pydata-sphinx-theme.js?digest=365ca57ee442770a23c6"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>